plan/logical_plan_builder.go

```diff
@@ -1960,16 +1986,35 @@ func (b *planBuilder) buildDelete(delete *ast.DeleteStmt) LogicalPlan {
 	setParentAndChildren(del, p)
 	del.SetSchema(expression.NewSchema())
 
+	var tableList []*ast.TableName
+	tableList = extractTableList(delete.TableRefs.TableRefs, tableList)
+
 	// Collect visitInfo.
 	if delete.Tables != nil {
 		// Delete a, b from a, b, c, d... add a and b.
-		for _, table := range delete.Tables.Tables {
-			b.visitInfo = appendVisitInfo(b.visitInfo, mysql.DeletePriv, table.Schema.L, table.TableInfo.Name.L, "")
+		dbName := b.ctx.GetSessionVars().CurrentDB
+		for _, tn := range delete.Tables.Tables {
+			foundMatch := false
+			for _, v := range tableList {
+				if dn := v.Schema.L; dn != "" {
+					dbName = dn
```

Comment:
If the first database name isn't empty and isn't the current database, and the next database name is empty, then the next database isn't the current database. Is there something wrong?